src/Symfony/Component/BrowserKit/HttpBrowser.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\BrowserKit;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpKernel\HttpCache\HttpCache;
+use Symfony\Component\HttpKernel\HttpCache\Store;
+use Symfony\Component\HttpKernel\HttpKernelBrowser;
+use Symfony\Component\HttpKernel\RealHttpKernel;
+use Symfony\Component\Mime\Part\AbstractPart;
+use Symfony\Component\Mime\Part\Multipart\FormDataPart;
+use Symfony\Component\Mime\Part\TextPart;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * An implementation of a browser using the HttpClient component
+ * to make real HTTP requests.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class HttpBrowser extends AbstractBrowser
```

Comment:
`WebBrowser` maybe?

Reply:
This consumes an `HttpClientInterface`, so `Http` makes sense to me - `Web` doesn't look as specific to me. Dunno what others think.

Reply:
I have no real pref :) just wondering if `HTTP Browser` is a "real world" thing ... (https://en.wikipedia.org/wiki/Web_browser vs. https://en.wikipedia.org/wiki/HTTP_browser)

Reply:
I tend to agree with @ro0NL here, especially as this class is optimized for browsing websites (and not APIs).

Reply:
@ro0NL HTTP qualifies this implementation of the Browser, as opposed to KernelBrowser and HttpKernelBrowser.

*all* browser implementations of BrowserKit are about web browsers (but we haven't added `web` everywhere in the component)